src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -609,6 +609,7 @@ internal static bool WriteConsideredUse(TypeSymbol type, BoundExpression value)
                         return WriteConsideredUse(null, boundConversion.Operand);
                     }
                 case BoundKind.DefaultOperator:
+                case BoundKind.ConvertedTupleLiteral:
```

Comment:
Ah, makes sense. Thanks for the clarification.

I have to say the message of the C# error does not convey this policy.

For the VB side, the policy seems different. It is really whether the variable is later used or left un-used (I didn't see any such warning suppression). Does that match your understanding?